docs/pull_request_template.md

```diff
@@ -0,0 +1,7 @@
+Closes OSD-1234
+
+# Proposed Changes
+*  
+* 
+* 
```

Comment:
I'd like to include a testing section to list what tests have been done / added to ensure it's a valid PR. 
Also a warning/checkbox to ensure this passes unit and integration tests.